src/raft_udp_serveripc.ml

```diff
@@ -46,46 +50,54 @@ end (* Pending_requests *)
   
 type connection_state = Pending_requests.t  
 
-type state = Raft_pb.state * connection_state 
+type state = RPb.state * connection_state * Compaction.handle  
 
 let initialize () = Pending_requests.empty
 
-let handle_notifications connection_state notifications = 
-
-  List.fold_left (fun acc notification ->
-
-    match notification with
-    | Raft_pb.Committed_data {Raft_pb.ids} -> 
-      List.fold_left (fun (connection_state, client_responses) id -> 
-          let connection_state, handle = Pending_requests.get connection_state id in 
-          match handle with
-          | None -> 
-            (connection_state, client_responses) 
-          | Some handle -> 
-            (connection_state, ((Pb.Add_log_success, handle)::client_responses))
-      ) acc ids 
-
-    | Raft_pb.New_leader _
-    | Raft_pb.No_leader  -> 
-        (* TODO need to go through all the pending requests. 
-         *)
-      acc
-  ) (connection_state, []) notifications
-
-
-let handle_raft_message ~logger ~stats ~now (raft_state, connection_state) msg = 
+let handle_notifications connection_state compaction_handle notifications = 
+
+  let connection_state, client_responses = List.fold_left (fun acc notification ->
+
+      match notification with
+      | RPb.Committed_data {RPb.rev_log_entries} -> 
+        let ids = List.map (fun ({RPb.id; _ }:RPb.log_entry) -> id) rev_log_entries in 
+        List.fold_left (fun (connection_state, client_responses) id -> 
+            let connection_state, handle = Pending_requests.get connection_state id in 
+            match handle with
+            | None -> 
+              (connection_state, client_responses) 
+            | Some handle -> 
+              (connection_state, ((Pb.Add_log_success, handle)::client_responses))
+        ) acc ids 
+
+      | RPb.New_leader _
+      | RPb.No_leader  -> 
+          (* TODO need to go through all the pending requests. 
+           *)
+        acc
+    ) (connection_state, []) notifications
+  in
+  Lwt_list.iter_s (function
```

Comment:
Initial testing shows that the throughput is affected by recording every log to file. This is expected, however we should experiment making this asynchronous. 

We could have a stream dedicated for commited data. The commited data would be pushed to the stream here (fast and synchronous) while a separate processing thread would iterate over the stream. This latter thread would record the log and push client responses.